File: python/mirage/commands/builtin/wandb/io.py
from mirage.core.wandb.read import read, read_stream
from mirage.core.wandb.readdir import readdir
from mirage.core.wandb.stat import stat
from mirage.vfs.adapter import VFSAdapter
from mirage.vfs.types import NativeReadOps, ReadOps

IO = VFSAdapter(read=ReadOps(readdir=readdir, read_bytes=read, stat=stat),
                native=NativeReadOps(read_stream=read_stream),
                is_mounted=lambda a: True,
                local=False,
                max_du_entries=1000).to_command_io()

File: typescript/packages/core/src/commands/builtin/wandb/io.ts
import { VFSAdapter } from '../../../vfs/adapter.ts'
import type { WandbAccessor } from '../../../accessor/wandb.ts'
import { read, readStream } from '../../../core/wandb/read.ts'
import { readdir } from '../../../core/wandb/readdir.ts'
import { stat } from '../../../core/wandb/stat.ts'
import type { CommandIO } from '../generic_bind/index.ts'
export const WANDB_IO: CommandIO<WandbAccessor> = new VFSAdapter<WandbAccessor>({
  read: { readdir, readBytes: read, stat },
  native: { readStream },
  isMounted: () => true,
  local: false,
  maxDuEntries: 1000,
}).toCommandIO()
